source/tools/glkernel-cmd/main.cpp

```diff
@@ -0,0 +1,56 @@
+#include "KernelGeneration.h"
+#include "KernelToJson.h"
+
+#include <iostream>
+#include <fstream>
+
+#include <cppassist/cmdline/ArgumentParser.h>
+
+#include <cppexpose/variant/Variant.h>
+#include <cppexpose/json/JSON.h>
+
+int main(int argc, char* argv[])
+{
+    cppassist::ArgumentParser argParser;
+    argParser.parse(argc, argv);
+
+    auto inFilename = argParser.value("--i");
+    auto outFilename = argParser.value("--o");
```

Comment:
A typical extension for command line tools is the integration in pipe architectures. We may specify an interface for this later.